src/storage/kv/transaction.rs

```diff
@@ -160,6 +160,14 @@ impl Transaction {
         Ok(())
     }
 
+    /// Adds a key-value pair to the store with the given timestamp.
+    pub fn set_at_ts(&mut self, key: &[u8], value: &[u8], ts: u64) -> Result<()> {
```

Comment:
Can we also add one unit test for this to check if it passes with different varying timestamps, and maybe what happens when an older timestamp is passed (past the previous committed ts) to check if it fails?

Reply:
> when an older timestamp is passed (past the previous committed ts) to check if it fails?

Why do we want to prevent this behaviour? I think it should be fine to be able to create a version record with the timestamp older that the existing one.

Reply:
I mean with a timestamp older than the last transaction committed timestamp. Because [this](https://github.com/surrealdb/vart/blob/d7d6fc52131853ef3656de6d4914cc47d4739268/src/art.rs#L1017) condition will fail right

Reply:
My bad, please ignore, got confused between version and ts

Reply:
No worries!